IPs/RSKIP419.md

```diff
@@ -0,0 +1,229 @@
+3. New composition commitment: m/n vote to commit new composition. New elections are disabled.
+4. Pre activation phase: immediately after commitment and during a number of blocks. No real change for end users.
+5. Activation phase: new Powpeg address activates, two-way peg starts operating from this new address.
+6. Migration phase: Funds from the retiring Powpeg wallet are moved to the active Powpeg.
+7. Post migration phase: Retiring Powpeg information is removed from the Bridge. New elections are re-enabled.
+
+The election process in itself doesn't change: the authorized members will still propose a new composition and once the voting finishes the commit event will be emitted, but the proposed Powpeg will not be committed yet and its validation will start.
+
+
+### New phase - Validation
+
+A new phase will be added, in between the _composition proposal_ and _new composition commitment_ phases, the _validation_ phase.
+During a certain number of blocks a new protocol will be started, Spendability Validation Protocol (SVP).
+
+If this protocol succeeds, the Bridge will progress to the next phase (_new composition commitment_).
+If this protocol fails, things will return to their previous state.
+
+
+### Spendability Validation Protocol (SVP)
+
+The Spendability Validation Protocol is a new process in which the current Powpeg will send funds to the newly elected Powpeg, for it to send them back, proving that it can truly hold and spend funds, before proceeding with the activation and migration of the funds in the wallet.
+
+This protocol involves three parts:
+- Proposed Powpeg funding: the proposed Powpeg won't have funds to create the proof so the current Powpeg will create an UTXO covering the expenses.
+- Proof transaction creation: once the proposed Powpeg receives the funds, it will create an UTXO to the current Powpeg to proof that it can spend them.
+- Proof transaction verification: the current Powpeg will receive the proposed Powpeg spending-proof transaction, validate it was created by the proposed Powpeg from the funding transaction and proceed with the commitment.
+
+#### Proposed Powpeg funding
+
+Once the SVP starts, the Bridge will send the required satoshis to the proposed Powpeg address. It should be enough for it to be able to send back a transaction. Since the size of the Proof transaction will vary depending on the size of the proposed Powpeg input script, the calculation of the funds sent from the current Powpeg should consider that.
+
+
+The process consists of the following 3 steps:
+
+##### 1. Funding transaction creation
+
+As soon as the proposed Powpeg has been voted (and saved in a new Bridge storage entry, key `proposedFederation`), the Bridge will create a SPV funding transaction for it.
+
+This transaction is similar to a peg-out where the recipient is the proposed Powpeg Bitcoin address, and the amount to receive is calculated based on the minimum amount of satoshis required to spend an UTXO by it.
```

Comment:
This value should probably be calculated as `minPegoutValue + bitcoin fees`. So that we ensure the proposed powpeg signs a transaction worth `minPegoutValue`.

It can be tricky to calculate this though, if bitcoin fees are oscilating during the process.